### PR TITLE
Better way to fetch data from multiple journals when ISSN is missing

### PR DESCRIPTION
**Pull request by Rekyt**
We still have maybe to find a solution for the `uniqueZetoc` case as shown in #10.
Tell me what you think ;)

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -66,21 +70,40 @@ parse_answer = function(api_answer, multiple = FALSE) {
...
+          journal_policy = tryCatch({
+            rr_journal_name(x["title"], check_key(key), qtype = "exact")
+          },
+          error = function(err) {
+            return(data.frame(title = x["title"],
+                              issn = x["issn"],
+                              preprint    = NA,
+                              postprint   = NA,
+                              pdf         = NA,
+                              romeocolour = NA))
```
Comment (Bisaloo) [inline R/utils.R]: Given the current master, this should also include the embargo policies before we merge it

File: R/utils.R [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
 #' recursively fetch data for each of one of them (`multiple = TRUE`) or
 #' return a data.frame containing only titles and ISSN of all matches
 #' (`multiple = FALSE`)
+#' @inheritParams check_key
 #'
 #' @keywords internal
 #'
 #' @import xml2
-parse_answer = function(api_answer, multiple = FALSE) {
+parse_answer = function(api_answer, multiple = FALSE, key = NULL) {
```
Comment (Bisaloo) [inline R/utils.R]: I not super happy with sending the key param back and forth between function but I haven't thought too much about it so maybe there is no other way...

Anyways, if you want to do this, you need to pass the `key` param when you call `parse_answer` in `rr_journal_issn` or `rr_journal_name`.

Comment (Rekyt) [reply R/utils.R]: Yep right... I know, it gets difficult to track the key parameter in this case.
However, it's a pity not to use it if the user has defined a key.
Especially when retrieving multiple journals.

**Comment by Rekyt** [comment]
We could return the last updated publisher policy by parsing the date of the last update. To have a criteria to prioritize.

However, in this case, I don't know which policy should be considered.

**Comment by Bisaloo** [comment]
As I said in http://www.sherpa.ac.uk/romeo/publishertypes.php?fIDnum=%7C&mode=simple&la=en&version=, there is a rule for priority. But I can't understand a thing from this document...

**Comment by Rekyt** [comment]
It doesn't match what is returned by the API.
However in the case of `0014-3855` the first returned publisher is `Elsevier Masson` with its parent indicated as `Elsevier` (which is the second returned publisher). So I would argue to return the policy of Elsevier Masson because it's more specific than the default policy of Elsevier.

**Comment by Bisaloo** [comment]
Hum, I'm not sure I agree (but I'm not sure I understand what you mean either). We should probably talk about this IRL some time soon.

**Comment by Bisaloo** [comment]
Okay, I think this is ready to merge now. We can improve the API key stuff later.

**Pull request merged**